### PR TITLE
feat: delete tenant route

### PR DESCRIPTION
**Pull request by danielgent**
Tested locally in auth-admin (as noted on ticket)

### CODE REVIEW

File: integration-test/management-api/tenants.spec.ts [hunk 1]
```diff
@@ -14,30 +15,19 @@ describe("tenants", () => {
...
-      headers: {
-        authorization: `Bearer ${token}`,
-      },
-    });
-
-    expect(response.status).toBe(200);
-
-    const body = (await response.json()) as Tenant[];
-    expect(body.length).toBe(0);
-  });
```
Comment (danielgent) [inline integration-test/management-api/tenants.spec.ts:30]: The existing tests had lots of duplicate behaivour.

Only the create user test is needed to
* check we get an empty list
* create a user
* check we get a populated list

File: integration-test/management-api/tenants.spec.ts [hunk 2]
```diff
@@ -57,52 +47,56 @@ describe("tenants", () => {
...
-    expect(body[0].id).toEqual(createdTenant.id);
+    expect(tenantsResponse2.status).toBe(200);
+    const body2 = (await tenantsResponse2.json()) as Tenant[];
+    expect(body2.length).toEqual(1);
+    expect(body2[0].id).toEqual(createdTenant.id);
   });
 
-  it("should get the first page of tenants", async () => {
+  it("should remove a tenant", async () => {
+    await setup(worker);
```
Comment (danielgent) [inline integration-test/management-api/tenants.spec.ts:63]: :eyes: this is the new test here *and* it uses `setup()` to seed some existing tenants :sunglasses:

File: src/adapters/interfaces/Tenants.ts [hunk 3]
```diff
@@ -15,4 +15,5 @@ export interface TenantsDataAdapter {
   get(id: string): Promise<Tenant | undefined>;
   list(params: ListParams): Promise<{ tenants: Tenant[]; totals?: Totals }>;
   update(id: string, tenant: Partial<Tenant>): Promise<void>;
+  remove(tenantId: string): Promise<boolean>;
```
Comment (markusahlstrand) [inline src/adapters/interfaces/Tenants.ts:18]: Ahh.. this was missing :)

**Review by markusahlstrand — APPROVED** [review]
LGTM!

File: src/adapters/interfaces/Tenants.ts (hunk 3 above)
Comment (danielgent) [reply src/adapters/interfaces/Tenants.ts:18]: Quite a few are missing on other routes. Happy to go through and add them :+1:

**Pull request merged**